include/realizations/catchment/Bmi_Module_Formulation.hpp

```diff
@@ -1128,6 +1131,10 @@ namespace realization {
                     //need to marshal data types to the reciever as well
                     //this could be done a little more elegantly if the provider interface were
                     //"type aware", but for now, this will do (but requires yet another copy)
+                    if(values.size() == 1){
+                        //FIXME this isn't generic broacasting, but works for scalar implementations
+                        values.resize(nbytes/varItemSize, values[0]);
+                    }
```

Comment:
These changes were incorporated in #556, so this PR needs to be rebased accordingly.